src/libflake/flake/lockfile.cc

```diff
@@ -108,8 +108,13 @@ LockFile::LockFile(
     const fetchers::Settings & fetchSettings,
     std::string_view contents, std::string_view path)
 {
-    auto json = nlohmann::json::parse(contents);
-
+    auto json = [=] {
+        try {
+            return nlohmann::json::parse(contents);
+        } catch (const nlohmann::json::parse_error & e) {
+            throw Error("While parsing '%s': %s", path, e.what());
```

Comment:
I had in mind an `addTrace` context when I put "while parsing" in the issue, but I think making it part of the final message is nicer, so let's make it sound a bit more natural:

```suggestion
            throw Error("Could not parse '%s': %s", path, e.what());
```

Reply:
Thanks! Updated.